java-frontend/src/main/java/org/sonar/java/Measurer.java

```diff
@@ -111,12 +112,20 @@ public void scanFile(JavaFileScannerContext context) {
 
     RangeDistributionBuilder fileComplexityDistribution = new RangeDistributionBuilder(LIMITS_COMPLEXITY_FILES);
     saveMetricOnFile(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, fileComplexityDistribution.add(fileComplexity).build());
+
+    if (isSonarQubeGreaterThanOrEqualTo63()) {
+      saveMetricOnFile(CoreMetrics.COGNITIVE_COMPLEXITY, context.compilationUnitCognitiveComplexity());
```

Comment:
`context.getTree()` provides access to `CompilationUnitTree`, so `CognitiveComplexityVisitor.cognitiveComplexity` can be used directly, thus avoiding the need for new method in the `JavaScannerContext` interface